packages/language-server/src/quick-fix.ts

```diff
@@ -0,0 +1,103 @@
+import { ExecuteCommandParams } from "vscode-languageserver";
+import {
+  Diagnostic,
+  Range as LSPRange,
+  TextDocument,
+  CodeAction,
+  Command,
+  CodeActionKind,
+  TextDocumentEdit,
+  TextEdit,
+} from "vscode-languageserver-types";
+import { getQuickFixIdInfo } from "@ui5-language-assistant/xml-views-quick-fix";
+import { parse, DocumentCstNode } from "@xml-tools/parser";
+import { buildAst, XMLDocument } from "@xml-tools/ast";
+import { LSPRangeToOffsetRange, offsetRangeToLSPRange } from "./range-utils";
+
+export function getQuickFixCodeAction(
+  document: TextDocument,
+  diagnostic: Diagnostic
+): CodeAction | undefined {
+  const documentText = document.getText();
+  // We prefer to parse the document again to avoid cache state handling
+  const { cst, tokenVector } = parse(documentText);
+  const xmlDocAst = buildAst(cst as DocumentCstNode, tokenVector);
+  switch (diagnostic.code) {
+    case 666: {
```

Comment:
we need a better diagnostic code number

Reply:
Examples of error status codes.
- https://github.com/Microsoft/TypeScript/blob/v1.6.2/src/compiler/diagnosticMessages.json

Reply:
The diagnostic code should be a cost exported from the validations package as a public API.